src/CourseDashboard.cpp

```diff
@@ -1,84 +1,36 @@
 #include "CourseDashboard.hpp"
-#include <stdexcept>
-#include <iostream>
-#include <algorithm>
+#include <fstream>
 
 CourseDashboard::CourseDashboard() {}
 
-void CourseDashboard::showAll()
+bool CourseDashboard::savingToFile(User & c)
 {
-    for (auto & user : users_)
+    std::ofstream out_file{ "../ParticipantsDetails.txt", std::ios::app };
+    if (!out_file)
     {
-        std::cout << user.getAllInfo();
+        std::cerr << "Error" << std::endl;
+        return false;
     }
+        out_file << c;
+        out_file.close();
+        return true;
 }
-
-void CourseDashboard::createUser(const User & user)
+bool CourseDashboard::readingFromFile()
 {
-    users_.emplace_back(user);
-}
-
-void CourseDashboard::deleteUserByNick(std::string nick)
-{  
-    auto it = std::find_if(std::begin(users_), std::end(users_), [nick](const auto & user)
+    std::ifstream in_file{"../ParticipantsDetails.txt"};
+    if (!in_file)
     {
-        return user.getNick() == nick;
-    });
-
-    if (it != std::end(users_))
-    {
-        users_.erase(it);
+       std::cerr << "Error" << std::endl;
+       return false;
     }
-}
-
-void CourseDashboard::retriveUserByNick(std::string nick)
-{
-    auto it = std::find_if(std::begin(users_), std::end(users_), [nick](const auto & user)
-    {
-        return user.getNick() == nick;
-    });
-    
-    if (it != std::end(users_))
+    std::string text{};
+    std::cout << "Reading from file ..." << std::endl;
+    std::cout << "=====================" << std::endl;
+    while (std::getline(in_file,text))
     {
-        std::cout << it->getAllInfo();
-    }
-}
-
-void CourseDashboard::updateUser(User & user)
-{
-    unsigned int number;
-    std::cout << user.getAllInfo();
-    std::cout << "Which data You want to edit? Give number: " << '\n';
-    std::cin >> number;
-    std::string name, nick, group, gitHub, firecode;
-    switch (number) {
-    case 1:
-        std::cout << "Give new value to data: ";
-        std::cin >> name;
-        user.setName(name);
-        break;
-    case 2:
-        std::cout << "Give new value to data: ";
-        std::cin >> nick;
-        user.setNick(nick);
-        break;
-    case 3:
-        std::cout << "Give new value to data: ";
-        std::cin >> group;
-        user.setGroup(group);
-        break;
-    case 4:
-        std::cout << "Give new value to data: ";
-        std::cin >> gitHub;
-        user.setGitHub(gitHub);
-        break;
-    case 5:
-        std::cout << "Give new value to data: ";
-        std::cin >> firecode;
-        user.setFirecode(firecode);
-        break;
-    default:
-        std::cout << "You gave wrong value.";
-        break;
+        std::cout << text << std::endl;
     }
+    std::cout << "=====================" << std::endl;
+    in_file.close();
+    return true;
```

Comment:
it doesn't read properly, does it? The data is not stored in vector of Users. It is only printed on the screen.